File: 086.LeetCode@530_Minimum_Absolute_Difference_in_BST/086.LeetCode@530_Minimum_Absolute_Difference_in_BST.py
class Solution:
    def __init__(self):
        self.min = float('inf')
        self.prev = float('inf')

    def help(self, root):
        if root is None:
            return
        self.help(root.left)
        self.min = min(self.min, abs(root.val - self.prev))
        self.prev = root.val
        self.help(root.right)

    def getMinimumDifference(self, root):
        self.help(root)
        return self.min

File: 086.LeetCode@530_Minimum_Absolute_Difference_in_BST/086.LeetCode@530_Minimum_Absolute_Difference_in_BST.java
class Solution {
    int min = Integer.MAX_VALUE, prev = Integer.MAX_VALUE;
    public void help(TreeNode root) {
        if (root == null)    return;
        help(root.left);
        min = Math.min(min, Math.abs(root.val - prev));
        prev = root.val;
        help(root.right);
    }
    public int getMinimumDifference(TreeNode root) {
        help(root);
        return min;
    }
}